src/metaopt/worker/trial.py

```diff
@@ -0,0 +1,244 @@
+# -*- coding: utf-8 -*-
+# pylint: skip-file
+"""
+:mod:`metaopt.worker.trial` -- Container class for `Trial` entity
+=================================================================
+
+.. module:: trial
+   :platform: Unix
+   :synopsis: Describe a particular training run, parameters and results
+
+"""
+
+import logging
+
+import six
+
+log = logging.getLogger(__name__)
+
+
+class Trial(object):
+    """Represents an entry in database/trials collection.
+
+    Attributes
+    ----------
+    experiment : str
+       Unique identifier for the experiment taht produced this trial.
+       Same as an `Experiment._id`.
+    status : str
+       Indicates how this trial is currently being used. Can take the following
+       values:
+
+       * 'new' : Denotes a fresh set of parameters suggested by an algorithm,
+          not yet tried out.
+       * 'reserved' : Indicates that this trial is currently being evaluated by
+          a worker process, it was a 'new' trial that got selected.
+       * 'suspended' : Means that an algorithm decided to stop the evaluation of
+          a 'reserved' trial prematurely.
+       * 'completed' : is the status of a previously 'reserved' trial that
+          successfully got evaluated. `Trial.results` must contain the evaluation.
+       * 'interrupted' : Indicates trials that are stopped from being evaluated
```

Comment:
Could an algo `interrupt` a Trial running and revive it again? Or in that case would be under `suspended`?

Or in this case an interrupted Trial by an algo would always be stated as `suspended`?

Nevermind it's clear enough from the comments. Just flow of thoughts from me :)

Reply:
Yes stopped trials from algorithms are always marked as suspended. Interruptions are caused by external factors, not the algorithm.